common/src/BasePlugin.php

```diff
@@ -218,4 +240,60 @@ private function isUniqueAttribute(array $array, string $value): bool {
     private function isValidGtin(string $value): bool {
         return preg_match('/^\d{8}(?:\d{4,6})?$/', $value);
     }
+
+    public function showUpdateNotice() {
+        $class = 'notice notice-info is-dismissible ' . $this->getUpdateNoticeClass();
+        $message = $this->getUpdateMessage();
+        if (!empty($message)) {
+            printf('<div class="%s">%s</div>', esc_attr($class), $message);
+        }
+    }
+
+    private function getUpdateMessage() {
+        $update_notices_file = WP_PLUGIN_DIR . DIRECTORY_SEPARATOR . $this->getSlug() . '/src/UpdateNotices.php';
+        if (file_exists($update_notices_file)) {
+            $updates = include $update_notices_file;
+            return $updates[$this->getPluginVersion($this->getSlug())] ?? null;
+        }
+        return null;
+    }
+
+    public function addUpdateNoticeDismissScript() {
+        $js_file = plugin_dir_url(__FILE__) . 'admin/js/update-notice.js';
+        $script_name = $this->getOptionName('notice_update');
+        wp_register_script(
+            $script_name,
+            $js_file
+        );
+        wp_localize_script($script_name, 'notice_params', [
+            'class' => $this->getUpdateNoticeClass(),
+            'hook' => $this->getUpdateNoticeDismissedAjaxHook(),
+        ]);
+        wp_enqueue_script($script_name);
+    }
+
+    private function getUpdateNoticeClass(): string {
+        return $this->getOptionName('custom_notice');
+    }
+
+    private function getUpdateNoticeDismissedAjaxHook(): string {
+        return $this->getOptionName('notice_dismiss');
+    }
+
+    public function dismissUpdateNotice() {
+        update_option(
+            $this->getOptionName('last_notice_version'),
+            $this->getPluginVersion($this->getSlug())
+        );
+    }
+
+    private function shouldDisplayUpdateNotice(): bool {
+        $last_notice_version = get_option($this->getOptionName('last_notice_version'));
+        return $last_notice_version
+            && version_compare(
```

Comment:
If the last notice version is not set (when they're updating from the previous version which doesn't have this), the notice should also be displayed! :)

Reply:
I did this by setting the default value of `get_option` to `''` , it would have still worked if the default value remained `false`, `(string) false` should be the same?